src/Lookup/EntityLookup.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use OutOfBoundsException;
+use Wikibase\DataModel\Entity\EntityDocument;
+use Wikibase\DataModel\Entity\EntityId;
+
+/**
+ * Service interface for retrieving Entities from storage.
+ *
+ * @since 1.1
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ * @author Daniel Kinzler
+ */
+interface EntityLookup {
+
+	/**
+	 * Returns the entity with the provided id or null if there is no such
+	 * entity.
+	 *
+	 * @note Implementations of this method may or may not resolve redirects.
+	 * Code that needs control over redirect resolution should use an
+	 * EntityRevisionLookup instead.
+	 *
+	 * @since 0.3
+	 *
+	 * @param EntityId $entityId
+	 *
+	 * @throws OutOfBoundsException
```

Comment:
Why and exception and the possibility of a returned null?? In case the exception is needed, having a more domain specific one, such as `EntityNotFoundException` would be nice. Did we not already have such an exception?